### PR TITLE
Make wrap fields also visible in entry editor

### LINKED ISSUE #4373: manually defined multiline fields in entry editor do not work
This is 
JabRef 5.0-dev--snapshot--2018-10-14--master--73ec32189
Linux 4.18.12-200.fc28.x86_64 amd64 
Java 1.8.0_172

In this and prior dev versions, I seem not be able (anymore) to define multiline fields in the entry editor. I used to have defined an own general field with multiline content; but now this is not shown as such any more. Only, I guess pre-defined, fields like "Abstract" are shown as multiline fields.

### PR DESCRIPTION
**Pull request by Siedlerchr**
Fixes #4373

I simple reused the FieldFormatterPreferences wrapFieldList. More like a hack.
The original multiline property got removed in #5230
A more elegant solution would be to maybe add this to the custom entry types dialog?

- [ ] Change in CHANGELOG.md described (if applicable)
- [ ] Tests created for changes (if applicable)
- [ ] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, submitted a pull request to the documentation repository.

### CODE REVIEW

File: src/main/java/org/jabref/preferences/JabRefPreferences.java [hunk 1]
```diff
@@ -571,7 +571,7 @@ private JabRefPreferences() {
         defaults.put(DEFAULT_BIBTEX_KEY_PATTERN, "[auth][year]");
         defaults.put(DO_NOT_RESOLVE_STRINGS_FOR, StandardField.URL.getName());
         defaults.put(RESOLVE_STRINGS_ALL_FIELDS, Boolean.FALSE);
-        defaults.put(NON_WRAPPABLE_FIELDS, "pdf;ps;url;doi;file;isbn;issn");
+        defaults.put(NON_WRAPPABLE_FIELDS, "pdf;ps;url;doi;file;isbn;issn;");
```
Comment (calixtus) [inline src/main/java/org/jabref/preferences/JabRefPreferences.java:574]: Is the additional `;` intended?
Does getStringList() return an additional empty list item then?

**Comment by Siedlerchr** [comment]
I would leave a proper implementation for the future. This depends also a bit on #6152

File: src/main/java/org/jabref/gui/fieldeditors/FieldEditors.java [hunk 2]
```diff
@@ -52,10 +51,10 @@ public static FieldEditorFX getForField(final Field field,
                 preferences.getBoolean(JabRefPreferences.ENFORCE_LEGAL_BIBTEX_KEY),
                 preferences.getBoolean(JabRefPreferences.ALLOW_INTEGER_EDITION_BIBTEX));
 
-        final boolean isSingleLine = FieldFactory.isSingleLineField(field);
+         boolean isMultiLine = FieldFactory.isMultiLineField(field, preferences.getFieldContentParserPreferences().getNonWrappableFields());
```
Comment (koppor) [inline src/main/java/org/jabref/gui/fieldeditors/FieldEditors.java:54]: Can we use an `enum` here? I really don't like the booleans - and we see changing the flag from `singleLine` to `multiLine` was necessary of readability. (Otherweise, the variable would be kept to flag `singleLine` - and negation would be used)

I propose to use a boolean enum as proposed at https://stackoverflow.com/a/12713550/873282

```java
enum USED_LINES {
    MULTI(true), SINGLE(false);
...
```

Maybe, just a simple enum would also be OK. Then, the construction from boolean would be a bit more tricky though.

File: src/main/java/org/jabref/gui/fieldeditors/PersonsEditor.java [hunk 3]
```diff
@@ -23,12 +23,10 @@ public PersonsEditor(final Field field,
...
                          final JabRefPreferences preferences,
                          final FieldCheckers fieldCheckers,
-                         final boolean isSingleLine) {
+                         final boolean isMultiLine) {
         this.viewModel = new PersonsEditorViewModel(field, suggestionProvider, preferences.getAutoCompletePreferences(), fieldCheckers);
 
-        textInput = isSingleLine
-                ? new EditorTextField()
-                : new EditorTextArea();
+        textInput = isMultiLine ? new EditorTextArea() : new EditorTextField();
```
Comment (koppor) [inline src/main/java/org/jabref/gui/fieldeditors/PersonsEditor.java:29]: Not that the true/false branches were just be wrapped. Another indicator for an enum!

File: src/main/java/org/jabref/gui/fieldeditors/FieldEditors.java (hunk 2 above)
Comment (tobiasdiez) [reply src/main/java/org/jabref/gui/fieldeditors/FieldEditors.java:54]: If you want to go in this direction, an enum `FieldType` with `Text` and `TextLong` would make the most sense and conveys the semantics. Can then be extended in the future to include `Keywords`, `Numeric` etc.

Comment (Siedlerchr) [reply src/main/java/org/jabref/gui/fieldeditors/FieldEditors.java:54]: I really would like to leave it for the moment and implement a more generic solution in a follow up PR. I think about integrating it in the custom entry types dialog.

**Comment by calixtus** [comment]
Ping.
How do we proceed here?

**Comment by Siedlerchr** [comment]
I m merge this now and will work on a more clean solution

**Pull request merged**